src/app/ReadClient.cpp

```diff
@@ -245,7 +245,19 @@ CHIP_ERROR ReadClient::SendReadRequest(ReadPrepareParams & aReadPrepareParams)
             }
         }
 
-        ReturnErrorOnFailure(request.IsFabricFiltered(aReadPrepareParams.mIsFabricFiltered).EndOfReadRequestMessage().GetError());
+        ReturnErrorOnFailure(request.IsFabricFiltered(aReadPrepareParams.mIsFabricFiltered).GetError());
+
+        if (aReadPrepareParams.mAttributePathParamsListSize != 0 && aReadPrepareParams.mpAttributePathParamsList != nullptr &&
+            aReadPrepareParams.mDataVersionFilterParamsListSize != 0 && aReadPrepareParams.mpDataVersionFilterParamsList != nullptr)
+        {
+            DataVersionFilterIBs::Builder & dataVersionFilterListBuilder = request.CreateDataVersionFilters();
+            ReturnErrorOnFailure(request.GetError());
+            ReturnErrorOnFailure(GenerateDataVersionFilterList(dataVersionFilterListBuilder,
+                                                               aReadPrepareParams.mpDataVersionFilterParamsList,
+                                                               aReadPrepareParams.mDataVersionFilterParamsListSize));
```

Comment:
Is there a reason we're not using a `Span<DataVersionFilter>` to store this state instead of separate point-and-length members?

Reply:
prefer span, would do the span switch for attributepathparamlist and eventpathparamlist and dataversionfilter together

Reply:
https://github.com/project-chip/connectedhomeip/issues/14915